streams-tests/shared/src/test/scala/zio/stream/experimental/ZSinkSpec.scala

```diff
@@ -540,14 +540,37 @@ object ZSinkSpec extends ZIOBaseSpec {
         //          )
         //        })
         //      ),
-        // testM("untilOutputZIO") {
-        //   val sink =
-        //     ZSink.head[Nothing, Int].untilOutputZIO(h => ZIO.succeed(h.fold(false)(_ >= 10)))
-        //   val assertions = ZIO.foreach(Chunk(1, 3, 7, 20)) { n =>
-        //     assertM(Stream.fromIterable(1 to 100).chunkN(n).run(sink))(equalTo(Some(Some(10))))
-        //   }
-        //   assertions.map(tst => tst.reduce(_ && _))
-        // },
+        testM("untilOutputZIO with head sink") {
+          val sink = ZSink.head[Nothing, Int].untilOutputZIO(h => ZIO.succeed(h.fold(false)(_ >= 10)))
+          val assertions = ZIO.foreach(Chunk(1, 3, 7, 20)) { n =>
+            assertM(ZStream.fromIterable(1 to 100).chunkN(n).run(sink))(equalTo(Some(Some(10))))
+          }
+          assertions.map(tst => tst.reduce(_ && _))
+        },
+        testM("untilOutputZIO failed predicate") {
+          assertM(
+            ZStream
+              .fromIterable(List(1, 2, 3, 4))
+              .run(ZSink.head[Nothing, Int].untilOutputZIO(_ => ZIO.fail("failed predicate")))
+          )(equalTo(None))
```

Comment:
This is incorrect; a failure from the predicate function should result in a failure of the channel.

Reply:
@iravid I reverted my latest commit, @jupposessho what exactly you mean with "handling the failure of the predicate" ?

Reply:
I mean, the predicate could fail for some value(since it's a `f => ZIO`, could be a http call or other effectual computation). See the current implementation of this method: `f(z).mapError(err => (Left(err), leftover))` ...

Reply:
I am sorry, but I still have no idea what you mean. Current behaviour is that If the predicate fails, channel fails as well. As indicated in the types, predicate fails with `OutErr1` which is also an err type of the channel. If I map over the error to some tuple of `(Either[OutErr1], Chunk[Chunk[L]])` (or whatever other value), code won't compile because compiler infers err type to be closest supertype of tuple and `OurErr1` which is `Any`. Only thing I can do in mapError is either `identity` or I can throw some kind of exception because then common supertype of `OutErr1` and `Nothing` is `OutErr1` but I don't think that's useful here. Anyway, thank you for your message and let me know what I am missing.